classes/pickup/pickup.gd

```diff
@@ -54,7 +54,12 @@ func assign_pickup_id(id) -> void:
 
 func pickup(body) -> void:
 	_award_pickup(body)
-	_pickup_sound()
+	
+	if respawn_seconds == 0.0:
+		_pickup_sound()
+	else:
+		sfx.play()
```

Comment:
This `if` check should be inside `_pickup_sound()`, not outside it.

(`_pickup_sound()` is supposed to be the _entire_ sound-playing routine. Having sound-playing logic _outside_ it makes the codebase fragile.)

Reply:
Good point, I'll fix this in a second.